src/Amalgam/string/StringManipulation.h

```diff
@@ -13,9 +13,16 @@ namespace StringManipulation
 	std::string NumberToString(size_t value);
 
 	//removes the first word from str and return the removed word
+	//if strip_word true, remove char_to_strip from beginning and end of returned word
 	std::string RemoveFirstWord(std::string &str);
 
-	std::vector<std::string> Split(std::string &s, char delim);
+	//splits a string by given delimiter
+	std::vector<std::string> Split(std::string &s, char delim = ' ');
+
+	//separates the argument string in a cross-platform manner and returns an appropriate vector of strings
+	//if non_greedy is true, the returned vector only contains a single string, and arg_string is modified
+	//to only contain the portion of the string after the removed section
+	std::vector<std::string> SplitArgString(std::string &arg_string, bool non_greedy = false);
```

Comment:
I think it makes more sense to make this `greedy` and default to true. I know originally said the other way but double negative is a bit ugly looking. Be sure to update the code too.